pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/RebalancePreCheckerResult.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.rebalance;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import javax.annotation.Nullable;
+
+
+/**
+ * Holds the pre-check result for each pre-check performed as part of RebalancePreChecker
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class RebalancePreCheckerResult {
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private final PreCheckStatus _preCheckStatus;
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private final String _message;
+
+  /**
+   * Constructor for RebalancePreCheckerResult
+   * @param preCheckStatus server related summary information
+   * @param message segment related summary information
+   */
+  @JsonCreator
+  public RebalancePreCheckerResult(@JsonProperty("preCheckStatus") PreCheckStatus preCheckStatus,
+      @JsonProperty("message") @Nullable String message) {
+    _preCheckStatus = preCheckStatus;
+    _message = message;
+  }
+
```

Comment:
```suggestion
  public static RebalancePreCheckerResult pass(@Nullable String msg) {
    return new RebalancePreCheckerResult(PreCheckStatus.PASS, msg);
  }
   public static RebalancePreCheckerResult warn(@Nullable String msg) {
    return new RebalancePreCheckerResult(PreCheckStatus.WARNING, msg);
  }
   public static RebalancePreCheckerResult error(@Nullable String msg) {
    return new RebalancePreCheckerResult(PreCheckStatus.ERROR, msg);
  }
```
Do you think adding methods like this helps simplify the return statements in pre-checkers? This way there's almost no need to set the variable `precheckStatus`

Reply:
done